### ISSUE TITLE
No index.html generated for website on chip demo

### ISSUE THREAD

**Issue by sbp-dev** [Mair/esp32-course#77]
Hi, I'm very new to web development. I followed all the instructions to install Svelte, Tailwind and DaisyUI. When I do `npm run dev` , site works fine. When I give `npm run build`, a build folder is generated under site but it has not index.html file so I can't read it from SPIFFS and return to the browser for rendering.
I tried `npm run preview`, site still seems to be working properly on localhost:3000

Attaching screenshot of contents of my site/build directory and assets fetched when hitting localhost:3000
Unable to proceed further, please advise!

![Svelte_1](https://user-images.githubusercontent.com/54526452/163713002-dd5d5e70-9b88-4cae-89fc-465957ebbfd3.PNG)
![Svelte_2](https://user-images.githubusercontent.com/54526452/163713005-d7d05801-50e5-4e13-92f5-7a9917489fbe.PNG)

**Comment by Mair** [comment]
hi @sbp-dev please post questions in the discussions https://github.com/Mair/esp32-course/discussions
The index should be there in the root of the build. A few things to check
1. the html files correlate to your svelte files, check you have an index.svelte in your route
2. are you using the @sveltejs/adapter-static
3. delete the build folder and rebuild it. it should be there 
4. failing that clone this repo and see if my code behaves the same way
this is my output

![image](https://user-images.githubusercontent.com/2477839/163716875-a63334be-c54c-4a12-9e8f-d2f270c8aa08.png)

**Comment by sbp-dev** [comment]
Hi Mair, thanks for the prompt response!
> please post questions in the discussions https://github.com/Mair/esp32-course/discussions

Sure, will keep that in mind for any future communication

I tried the following:

- Cloned your repository
- cd into `_15_Internet_Server/v14/_11_web_site_10_change_networks/site`
- `rm -r build/`
- `npm install`
- `npm i @sveltejs/adapter-static`
- `npm run build`
- `ls -alh build/`

Still no .html files! The site works locally though if I give `npm run dev` / `npm run preview`
Also, tried with `npm i @sveltejs/adapter-static --save` and `npm i @sveltejs/adapter-static@next --save` with same results

Please find attached console logs. Thanks in advance.

```
subha@SHADOWBOX MINGW64 /d/Work/embedded/ESP32/projects/esp32-course/_15_Internet_Server/v14/_11_web_site_10_change_networks/site (master)
$ rm -r build/

subha@SHADOWBOX MINGW64 /d/Work/embedded/ESP32/projects/esp32-course/_15_Internet_Server/v14/_11_web_site_10_change_networks/site (master)
$ ls
node_modules/  package.json  package-lock.json  postcss.config.cjs  README.md  src/  static/  svelte.config.js  tailwind.config.cjs  tsconfig.json

subha@SHADOWBOX MINGW64 /d/Work/embedded/ESP32/projects/esp32-course/_15_Internet_Server/v14/_11_web_site_10_change_networks/site (master)
$ git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean

subha@SHADOWBOX MINGW64 /d/Work/embedded/ESP32/projects/esp32-course/_15_Internet_Server/v14/_11_web_site_10_change_networks/site (master)
$ npm install
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-freebsd-64@0.14.36 (node_modules\esbuild\node_modules\esbuild-freebsd-64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-freebsd-64@0.14.36: wanted {"os":"freebsd","arch":"x64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-android-arm64@0.14.36 (node_modules\esbuild\node_modules\esbuild-android-arm64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-android-arm64@0.14.36: wanted {"os":"android","arch":"arm64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-darwin-64@0.14.36 (node_modules\esbuild\node_modules\esbuild-darwin-64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-darwin-64@0.14.36: wanted {"os":"darwin","arch":"x64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-darwin-arm64@0.14.36 (node_modules\esbuild\node_modules\esbuild-darwin-arm64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-darwin-arm64@0.14.36: wanted {"os":"darwin","arch":"arm64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-arm@0.14.36 (node_modules\esbuild\node_modules\esbuild-linux-arm):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-arm@0.14.36: wanted {"os":"linux","arch":"arm"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-ppc64le@0.14.36 (node_modules\esbuild\node_modules\esbuild-linux-ppc64le):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-ppc64le@0.14.36: wanted {"os":"linux","arch":"ppc64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-android-64@0.14.36 (node_modules\esbuild\node_modules\esbuild-android-64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-android-64@0.14.36: wanted {"os":"android","arch":"x64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-arm64@0.14.36 (node_modules\esbuild\node_modules\esbuild-linux-arm64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-arm64@0.14.36: wanted {"os":"linux","arch":"arm64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-32@0.14.36 (node_modules\esbuild\node_modules\esbuild-linux-32):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-32@0.14.36: wanted {"os":"linux","arch":"ia32"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-64@0.14.36 (node_modules\esbuild\node_modules\esbuild-linux-64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-64@0.14.36: wanted {"os":"linux","arch":"x64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-s390x@0.14.36 (node_modules\esbuild\node_modules\esbuild-linux-s390x):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-s390x@0.14.36: wanted {"os":"linux","arch":"s390x"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-netbsd-64@0.14.36 (node_modules\esbuild\node_modules\esbuild-netbsd-64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-netbsd-64@0.14.36: wanted {"os":"netbsd","arch":"x64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-openbsd-64@0.14.36 (node_modules\esbuild\node_modules\esbuild-openbsd-64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-openbsd-64@0.14.36: wanted {"os":"openbsd","arch":"x64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-sunos-64@0.14.36 (node_modules\esbuild\node_modules\esbuild-sunos-64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-sunos-64@0.14.36: wanted {"os":"sunos","arch":"x64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-windows-32@0.14.36 (node_modules\esbuild\node_modules\esbuild-windows-32):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-windows-32@0.14.36: wanted {"os":"win32","arch":"ia32"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-windows-arm64@0.14.36 (node_modules\esbuild\node_modules\esbuild-windows-arm64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-windows-arm64@0.14.36: wanted {"os":"win32","arch":"arm64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-freebsd-arm64@0.14.36 (node_modules\esbuild\node_modules\esbuild-freebsd-arm64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-freebsd-arm64@0.14.36: wanted {"os":"freebsd","arch":"arm64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-mips64le@0.14.36 (node_modules\esbuild\node_modules\esbuild-linux-mips64le):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-mips64le@0.14.36: wanted {"os":"linux","arch":"mips64el"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-riscv64@0.14.36 (node_modules\esbuild\node_modules\esbuild-linux-riscv64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-riscv64@0.14.36: wanted {"os":"linux","arch":"riscv64"} (current: {"os":"win32","arch":"x64"})
npm WARN site@0.0.1 No repository field.
npm WARN site@0.0.1 No license field.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

updated 1 package and audited 184 packages in 19.544s

21 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities


subha@SHADOWBOX MINGW64 /d/Work/embedded/ESP32/projects/esp32-course/_15_Internet_Server/v14/_11_web_site_10_change_networks/site (master)
$ npm i @sveltejs/adapter-static
npm WARN site@0.0.1 No repository field.
npm WARN site@0.0.1 No license field.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-android-64@0.14.36 (node_modules\esbuild-android-64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-android-64@0.14.36: wanted {"os":"android","arch":"x64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-freebsd-64@0.14.36 (node_modules\esbuild-freebsd-64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-freebsd-64@0.14.36: wanted {"os":"freebsd","arch":"x64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-android-arm64@0.14.36 (node_modules\esbuild-android-arm64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-android-arm64@0.14.36: wanted {"os":"android","arch":"arm64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-32@0.14.36 (node_modules\esbuild-linux-32):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-32@0.14.36: wanted {"os":"linux","arch":"ia32"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-darwin-64@0.14.36 (node_modules\esbuild-darwin-64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-darwin-64@0.14.36: wanted {"os":"darwin","arch":"x64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-arm@0.14.36 (node_modules\esbuild-linux-arm):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-arm@0.14.36: wanted {"os":"linux","arch":"arm"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-darwin-arm64@0.14.36 (node_modules\esbuild-darwin-arm64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-darwin-arm64@0.14.36: wanted {"os":"darwin","arch":"arm64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-64@0.14.36 (node_modules\esbuild-linux-64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-64@0.14.36: wanted {"os":"linux","arch":"x64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-ppc64le@0.14.36 (node_modules\esbuild-linux-ppc64le):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-ppc64le@0.14.36: wanted {"os":"linux","arch":"ppc64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-freebsd-arm64@0.14.36 (node_modules\esbuild-freebsd-arm64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-freebsd-arm64@0.14.36: wanted {"os":"freebsd","arch":"arm64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-arm64@0.14.36 (node_modules\esbuild-linux-arm64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-arm64@0.14.36: wanted {"os":"linux","arch":"arm64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-mips64le@0.14.36 (node_modules\esbuild-linux-mips64le):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-mips64le@0.14.36: wanted {"os":"linux","arch":"mips64el"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-riscv64@0.14.36 (node_modules\esbuild-linux-riscv64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-riscv64@0.14.36: wanted {"os":"linux","arch":"riscv64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-netbsd-64@0.14.36 (node_modules\esbuild-netbsd-64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-netbsd-64@0.14.36: wanted {"os":"netbsd","arch":"x64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-openbsd-64@0.14.36 (node_modules\esbuild-openbsd-64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-openbsd-64@0.14.36: wanted {"os":"openbsd","arch":"x64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-linux-s390x@0.14.36 (node_modules\esbuild-linux-s390x):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-linux-s390x@0.14.36: wanted {"os":"linux","arch":"s390x"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-sunos-64@0.14.36 (node_modules\esbuild-sunos-64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-sunos-64@0.14.36: wanted {"os":"sunos","arch":"x64"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-windows-32@0.14.36 (node_modules\esbuild-windows-32):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-windows-32@0.14.36: wanted {"os":"win32","arch":"ia32"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: esbuild-windows-arm64@0.14.36 (node_modules\esbuild-windows-arm64):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for esbuild-windows-arm64@0.14.36: wanted {"os":"win32","arch":"arm64"} (current: {"os":"win32","arch":"x64"})

+ @sveltejs/adapter-static@1.0.0-next.29
updated 1 package and audited 184 packages in 6.151s

22 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities


subha@SHADOWBOX MINGW64 /d/Work/embedded/ESP32/projects/esp32-course/_15_Internet_Server/v14/_11_web_site_10_change_networks/site (master)
$ npm run build

> site@0.0.1 build D:\Work\embedded\ESP32\projects\esp32-course\_15_Internet_Server\v14\_11_web_site_10_change_networks\site
> svelte-kit build

Your tsconfig.json should extend the configuration generated by SvelteKit:
{
  "extends": "./.svelte-kit/tsconfig.json"
}
vite v2.9.5 building for production...
transforming...

🌼 daisyUI components 2.1.0  https://github.com/saadeghi/daisyui
  ✔︎ Including:  base, components, themes[26], utilities

✓ 17 modules transformed.
rendering chunks...
.svelte-kit/output/client/_app/manifest.json                               1.98 KiB
.svelte-kit/output/client/_app/pages/__layout.svelte-5d5e62f8.js           2.55 KiB / gzip: 1.25 KiB
.svelte-kit/output/client/_app/error.svelte-97a12756.js                    1.56 KiB / gzip: 0.75 KiB
.svelte-kit/output/client/_app/pages/ap-to-sta.svelte-d2fd9c18.js          3.85 KiB / gzip: 1.87 KiB
.svelte-kit/output/client/_app/pages/index.svelte-28eeab61.js              1.61 KiB / gzip: 0.84 KiB
.svelte-kit/output/client/_app/pages/toggle-led.svelte-5b5de70c.js         1.21 KiB / gzip: 0.62 KiB
.svelte-kit/output/client/_app/pages/ws-button-count.svelte-712478dc.js    1.71 KiB / gzip: 0.97 KiB
.svelte-kit/output/client/_app/chunks/index-91fd57d4.js                    7.40 KiB / gzip: 3.01 KiB
.svelte-kit/output/client/_app/start-5ae3a68b.js                           23.43 KiB / gzip: 8.73 KiB
.svelte-kit/output/client/_app/assets/pages/__layout.svelte-87e18bca.css   49.32 KiB / gzip: 8.57 KiB
vite v2.9.5 building SSR bundle for production...
transforming...
✓ 16 modules transformed.
Generated an empty chunk: "hooks"
rendering chunks...
.svelte-kit/output/server/manifest.json                             1.89 KiB
.svelte-kit/output/server/index.js                                  65.74 KiB
.svelte-kit/output/server/entries/endpoints/api/_route_.ts.js       0.69 KiB
.svelte-kit/output/server/entries/pages/__layout.svelte.js          1.01 KiB
.svelte-kit/output/server/entries/fallbacks/error.svelte.js         0.72 KiB
.svelte-kit/output/server/entries/pages/ap-to-sta.svelte.js         1.06 KiB
.svelte-kit/output/server/entries/pages/index.svelte.js             0.75 KiB
.svelte-kit/output/server/entries/pages/toggle-led.svelte.js        0.31 KiB
.svelte-kit/output/server/entries/pages/ws-button-count.svelte.js   0.81 KiB
.svelte-kit/output/server/chunks/index-7340f048.js                  3.10 KiB
.svelte-kit/output/server/chunks/hooks-1c45ba0b.js                  0.00 KiB

Run npm run preview to preview your production build locally.

> Using @sveltejs/adapter-static
  You should set `config.kit.prerender.default` to `true` if no fallback is specified
  Wrote site to "build"
  ✔ done

subha@SHADOWBOX MINGW64 /d/Work/embedded/ESP32/projects/esp32-course/_15_Internet_Server/v14/_11_web_site_10_change_networks/site (master)
$ ls -alh build/
total 12K
drwxr-xr-x 1 subha 197609    0 Apr 17 20:26 ./
drwxr-xr-x 1 subha 197609    0 Apr 17 20:26 ../
drwxr-xr-x 1 subha 197609    0 Apr 17 20:26 _app/
-rw-r--r-- 1 subha 197609 1.6K Apr 17 18:20 favicon.png
```

**Comment by Mair** [comment]
@sbp-dev interesting
have you followed these steps in https://learnesp32.com/videos/15/1_introduction%20to%20svelte%2C%20tailwind%20and%20daisy-ui at 7:40?

**Comment by Mair** [comment]
@sbp-dev Ok got it.
This is a recent change made to svelte kit.
In your svelte.config.js under kit add `prerender: { default: true },`

It should look like this
```js
import adapter from "@sveltejs/adapter-static";
import preprocess from "svelte-preprocess";

/** @type {import('@sveltejs/kit').Config} */
const config = {
  // Consult https://github.com/sveltejs/svelte-preprocess
  // for more information about preprocessors
  preprocess: preprocess(),

  kit: {
    adapter: adapter(),
    prerender: { default: true },
  },
};

export default config;
```

Then a re-build and you should be good.

**Comment by Mair** [comment]
I'll update the video 
merged #79

**Comment by sbp-dev** [comment]
Perfect, this works now!
Thanks again for the quick help.
